### DISCUSSION TITLE
How to enable async mode for insert using SQL via tcp connection?

### DISCUSSION THREAD

**Discussion by samgumirov** [ClickHouse/ClickHouse#33722] [Q&A] [answered]
Hello community and CH team,
Can I use insert async mode with SQL? I found only mention to use `async_insert` (https://clickhouse.com/blog/en/2021/clickhouse-v21.11-released/) for inserts and one example of using via HTTP API. I currently use https://github.com/mymarilyn/clickhouse-driver to `INSERT` from S3 bucket. I currently use aws lambda (timeout is 30 secs), so it would be great to have async inserts.
Thanks for any feedback!

**Comment by den-crane** [comment, marked as answer]
async_insert inserts are not related to `INSERT from S3`. 
async_insert is to insert data from the client not form S3.

Clickhouse has no feature to insert data asynchronously from S3 (fire & forget mode)
